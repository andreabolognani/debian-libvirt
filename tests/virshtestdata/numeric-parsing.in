echo Test a regular numeric option

echo Non-numeric value
cpu-stats test --start abc

echo Numeric value with invalid suffix
cpu-stats test --start 42WB

echo Numeric value with valid suffix. Suffixes are not supported for
echo regular numeric options, so this value is rejected
cpu-stats test --start 42MB

echo Numeric value bigger than INT_MAX
cpu-stats test --start 2147483648

echo Negative numeric value. The value is not valid for the command
echo we are testing, but it has been parsed correctly
cpu-stats test --start -1

echo Test a scaled numeric option

echo Non-numeric value
setmaxmem test abc

echo Numeric value with invalid suffix
setmaxmem test 42WB

echo Numeric value with valid suffix
setmaxmem test 42MB --config

echo Numeric value bigger than INT_MAX. No failure here because
echo scaled numeric values are unsigned long long
setmaxmem test 2147483648 --config

echo Negative numeric value
setmaxmem test -1

echo Zero. The value is not valid for the command were testing, but
echo it has been parsed correctly
setmaxmem test 0

echo Numeric value
setmaxmem test 42 --config
