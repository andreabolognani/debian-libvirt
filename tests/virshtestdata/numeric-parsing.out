Test a regular numeric option
Non-numeric value
error: Numeric value 'abc' for <start> option is malformed or out of range

Numeric value with invalid suffix
error: Numeric value '42WB' for <start> option is malformed or out of range

Numeric value with valid suffix. Suffixes are not supported for
regular numeric options, so this value is rejected
error: Numeric value '42MB' for <start> option is malformed or out of range

Numeric value bigger than INT_MAX
error: Numeric value '2147483648' for <start> option is malformed or out of range

Negative numeric value. The value is not valid for the command
we are testing, but it has been parsed correctly
error: Invalid value for start CPU

Test a scaled numeric option
Non-numeric value
error: Scaled numeric value 'abc' for <size> option is malformed or out of range

Numeric value with invalid suffix
error: Scaled numeric value '42WB' for <size> option is malformed or out of range
error: invalid argument: unknown suffix 'WB'

Numeric value with valid suffix

Numeric value bigger than INT_MAX. No failure here because
scaled numeric values are unsigned long long

Negative numeric value
error: Scaled numeric value '-1' for <size> option is malformed or out of range

Zero. The value is not valid for the command were testing, but
it has been parsed correctly
error: Unable to change MaxMemorySize
error: memory in virDomainSetMemoryFlags must not be zero

Numeric value

